packages/medusa-plugin-add-ons/src/api/routes/admin/update-add-on.js

```diff
@@ -11,21 +11,21 @@ export default async (req, res) => {
         amount: Validator.number().required(),
       })
       .optional(),
-    valid_for: Validator.array().items(Validator.string()).optional(),
+    valid_for: Validator.array().optional(),
     metadata: Validator.object().optional(),
```

Comment:
Metadata can't be updated in the `addOnService.update`. 

In the controller check if there is metadata in the request if so use:
```js
Object.entries(value.metadata).map(([key, value]) => {
  addOnService.setMetadata(id, key, value)
})
```
finally do `delete value.metadata` before calling `addOnService.update`.